apps/smart-contracts/core/test/fixtures/UniswapFixtures.ts

```diff
@@ -1,8 +1,21 @@
 import { FakeContract, smock } from '@defi-wonderland/smock'
+import { abi as UNIV3_FACTORY_ABI } from '@uniswap/v3-core/artifacts/contracts/UniswapV3Factory.sol/UniswapV3Factory.json'
+import { abi as UNIV3_POOL_ABI } from '@uniswap/v3-core/artifacts/contracts/UniswapV3Pool.sol/UniswapV3Pool.json'
 import { abi as SWAP_ROUTER_ABI } from '@uniswap/v3-periphery/artifacts/contracts/SwapRouter.sol/SwapRouter.json'
-import { BaseContract } from 'ethers'
+import { BaseContract, Contract } from 'ethers'
+import { ethers } from 'hardhat'
 
 export async function fakeSwapRouterFixture(): Promise<FakeContract<BaseContract>> {
   const fakeContract = await smock.fake(SWAP_ROUTER_ABI)
   return fakeContract
 }
+
+export async function attachUniV3Factory(address: string): Promise<Contract> {
+  const contract = await ethers.getContractAt(UNIV3_FACTORY_ABI, address)
+  return contract
```

Comment:
```suggestion
  return await ethers.getContractAt(UNIV3_FACTORY_ABI, address)
```
?
(maybe need brackets around the await part)

same for other functions

Reply:
Ok yeah I figured out what was happening here, we don't need to `await` the expression here and this doesn't need to be labeled `async` since we are just directly returning the promise for fakes.